src/physlr-molecules.cc

```diff
@@ -0,0 +1,398 @@
+#include <fstream>
+#include <getopt.h>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/biconnected_components.hpp>
+#include <boost/graph/graph_utility.hpp>
+#include <boost/graph/subgraph.hpp>
+
+#if _OPENMP
+#include <omp.h>
+#endif
+
+#define PROGRAM "physlr-molecules"
+#define PACKAGE_NAME "physlr"
+#define GIT_REVISION "pre-autotools"
+
+static int
+memory_usage()
+{
+	int mem = 0;
+	std::ifstream proc("/proc/self/status");
+	std::string s;
+	while (getline(proc, s), !proc.fail()) {
+		if (s.substr(0, 6) == "VmSize") {
+			std::stringstream convert(s.substr(s.find_last_of('\t'), s.find_last_of('k') - 1));
+			if (!(convert >> mem)) {
+				return 0;
+			}
+			return mem;
+		}
+	}
+	return mem;
+}
+
+struct vertexProperties
+{
+	std::string name = "";
+	int weight = 0;
+	size_t indexOriginal = 0;
+};
+
+struct edgeProperties
+{
+	int weight = 0;
+};
+
+struct edgeComponent_t
+{
+	enum
+	{
+		num = 555
+	};
+	using kind = boost::edge_property_tag;
+} edgeComponent;
+
+using graph_t = boost::subgraph<boost::adjacency_list<
+    boost::vecS,
+    boost::vecS,
+    boost::undirectedS,
+    vertexProperties,
+    boost::property<
+        boost::edge_index_t,
+        int,
+        boost::property<edgeComponent_t, std::size_t, edgeProperties>>>>;
+using vertex_t = graph_t::vertex_descriptor;
+using edge_t = graph_t::edge_descriptor;
+using barcodeToIndex_t = std::unordered_map<std::string, vertex_t>;
+using indexToBarcode_t = std::unordered_map<vertex_t, std::string>;
+using vertexSet_t = std::unordered_set<vertex_t>;
+using componentToVertexSet_t = std::vector<vertexSet_t>;
+using vertexToComponent_t = std::unordered_map<vertex_t, size_t>;
+using vecVertexToComponent_t = std::vector<vertexToComponent_t>;
+
+static void
+printVersion()
+{
+	const char VERSION_MESSAGE[] =
+	    PROGRAM " (" PACKAGE_NAME ") " GIT_REVISION "\n"
+	            "Written by Johnathan Wong.\n"
+	            "\n"
+	            "Copyright 2019 Canada's Michael Smith Genome Science Centre\n";
+	std::cerr << VERSION_MESSAGE << std::endl;
+	exit(EXIT_SUCCESS);
+}
+
+static void
+printErrorMsg(const std::string& progname, const std::string& msg)
+{
+	std::cerr << progname << ": " << msg
+	          << "\nTry 'physlr-molecules --help' for more information.\n";
+}
+
+static void
+printUsage(const std::string& progname)
+{
+	std::cout << "Usage:  " << progname
+	          << "  [-s SEPARATION-STRATEGY] [-v] FILE...\n\n"
+	             "  -v         enable verbose output\n"
+	             "  -s --separation-strategy   \n"
+	             "  --help     display this help and exit\n"
+	             "  SEPARATION-STRATEGY      `+` separated list of molecule separation strategies "
```

Comment:
I think I suggested before that the separation strategy help information would be best to be by the `-s`?

Reply:
I must have missed it if you did. It's done now.